service-base/src/main/java/org/eclipse/hono/service/auth/HonoSaslAuthenticatorFactory.java

```diff
@@ -37,13 +37,33 @@
      * Verifies credentials by means of sending authentication requests to address
      * {@link AuthenticationConstants#EVENT_BUS_ADDRESS_AUTHENTICATION_IN} on the Vert.x
      * Event Bus.
-     * 
+     *
      * @param vertx the Vertx environment to run the factory in.
      * @param validator The object to use for validating auth tokens.
+     * @param actualAuthenticationService The authentication service that will handle the authentication requests sent
+     *            on the Vert.x event bus. Used for determining the supported SASL mechanisms.
      * @throws NullPointerException if any of the parameters is {@code null}.
      */
     @Autowired
-    public HonoSaslAuthenticatorFactory(final Vertx vertx, @Qualifier(AuthenticationConstants.QUALIFIER_AUTHENTICATION) final AuthTokenHelper validator) {
+    public HonoSaslAuthenticatorFactory(final Vertx vertx,
+            @Qualifier(AuthenticationConstants.QUALIFIER_AUTHENTICATION) final AuthTokenHelper validator,
+            final AuthenticationService actualAuthenticationService) {
+        this(new EventBusAuthenticationService(vertx, validator,
+                actualAuthenticationService.getSupportedSaslMechanisms()));
+    }
+
+    /**
+     * Creates a new factory for a Vertx environment.
+     * <p>
+     * Verifies credentials by means of sending authentication requests to address
+     * {@link AuthenticationConstants#EVENT_BUS_ADDRESS_AUTHENTICATION_IN} on the Vert.x
+     * Event Bus.
+     *
+     * @param vertx the Vertx environment to run the factory in.
+     * @param validator The object to use for validating auth tokens.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    public HonoSaslAuthenticatorFactory(final Vertx vertx, final AuthTokenHelper validator) {
```

Comment:
FMPOV this should go *before* the new constructor variant. Is there a particular reason why the qualifier has been removed from the *validator* parameter?

Reply:
> Is there a particular reason why the qualifier has been removed from the validator parameter?

FMPOV, the qualifier was included in connection with the `Autowired` annotation. Since the newly introduced constructor has the `Autowired` annotation now, I've removed the `Autowired` annotation here along with the qualifier for the _validator_ parameter.

Reply:
Ah, I see. So this constructor will not be subject to auto wiring at all.